cli/azd/internal/appdetect/spring_boot.go

```diff
@@ -495,21 +494,21 @@ func logMetadataUpdated(info string) {
 	log.Printf("Metadata updated. %s.", info)
 }
 
-func detectSpringBootVersion(currentRoot *mavenProject, mavenProject *mavenProject) string {
-	// mavenProject prioritize than rootProject
-	if mavenProject != nil {
-		if version := detectSpringBootVersionFromProject(mavenProject); version != UnknownSpringBootVersion {
+func detectSpringBootVersion(currentRoot *pom, parentPom *pom) string {
+	// currentRoot prioritize than parentPom
+	if currentRoot != nil {
+		if version := detectSpringBootVersionFromProject(currentRoot); version != UnknownSpringBootVersion {
 			return version
 		}
 	}
-	// fallback to detect root project
-	if currentRoot != nil {
-		return detectSpringBootVersionFromProject(currentRoot)
+	// fallback to detect parentPom
+	if parentPom != nil {
+		return detectSpringBootVersionFromProject(parentPom)
```

Comment:
This will be deleted in next PR, because I plan to change pom to effective pom immediately after unmarshall from file. So there is no necessary to take parentPom anymore.